src/chainparams.cpp

```diff
@@ -298,3 +320,24 @@ bool SelectParamsFromCommandLine()
     SelectParams(network);
     return true;
 }
+
+
+// Block height must be >0 and less than the halving interval block height
+// Index variable i ranges from 0 - (vFoundersRewardAddress.size()-1)
+std::string CChainParams::GetFoundersRewardAddress(size_t nHeight) const
+{
+    assert(nHeight>0 && nHeight<consensus.nSubsidyHalvingInterval);
+    size_t i = (size_t)floor((double(nHeight)/consensus.nSubsidyHalvingInterval)*vFoundersRewardAddress.size());
```

Comment:
It seems like there is some inconsistency in the duration of the Founders' Reward. I had been under the impression that the Founders' Reward lasted exactly until the first halvening. But the first halvening is at block ``consensusParams.SubsidySlowStartShift() + consensusParams.nSubsidyHalvingInterval``, [implemented here](https://github.com/zcash/zcash/blob/master/src/main.cpp#L1386), whereas the Founders' Reward is enforced for ``(nHeight > 0) && (nHeight < consensusParams.nSubsidyHalvingInterval)``, [implemented here](https://github.com/zcash/zcash/blob/master/src/main.cpp#L3089). This seems wrong; I think the latter should be ``(nHeight > 0) && (nHeight < consensusParams.SubsidySlowStartShift() + consensusParams.nSubsidyHalvingInterval)``.

Also, I don't like the use of floating point here.

I would suggest adding a function ``consensus.GetLastFoundersRewardBlockHeight()``, and then writing it like this:
```
assert(nHeight > 0 && nHeight <= consensus.GetLastFoundersRewardBlockHeight());
size_t addressChangeInterval = (consensus.GetLastFoundersRewardBlockHeight() + vFoundersRewardAddress.size() - 1)/vFoundersRewardAddress.size();
size_t i = (nHeight-1) / addressChangeInterval;
return vFoundersRewardAddress[i];
```